sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/PostLedgerEntryOperation.cs

```diff
@@ -55,12 +55,37 @@ protected PostLedgerEntryOperation()
 
         async ValueTask<OperationState> IOperation.UpdateStateAsync(bool async, CancellationToken cancellationToken)
         {
-            var statusResponse = async
-                ? await _client.GetTransactionStatusAsync(
-                        Id,
-                        new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow })
-                    .ConfigureAwait(false)
-                : _client.GetTransactionStatus(Id, new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow });
+            int retryCount = 0;
+            Azure.Response statusResponse = null;
+            while (retryCount < 3)
+            {
+                 statusResponse = async
+                    ? await _client.GetTransactionStatusAsync(
+                            Id,
+                            new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow })
+                        .ConfigureAwait(false)
+                    : _client.GetTransactionStatus(Id, new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow });
+
+                // The transaction may not be found due to unexpected loss of session stickiness.
+                // This may occur when the connected node changes and transactions have not been fully replicated.
+                // We will perform retry logic to ensure that we have waited for the transactions to fully replicate before throwing an error.
+                if (statusResponse.Status == (int)HttpStatusCode.NotFound)
+                {
+                    ++retryCount;
+                }
+                else
+                {
+                    break;
+                }
+
+                // Add a 0.5 second delay between retries.
+                int delayTimeMs = 500;
+                if (async) {
+                    await Task.Delay(delayTimeMs).ConfigureAwait(false);
+                } else {
+                    Thread.Sleep(delayTimeMs);
```

Comment:
Thread.Sleep does not play well with all SynchronizationContexts.  Please don't merge this as-is.  Ideally we'd be able to make this work with a custom DelayStrategy instead.

Reply:
Based off of our email thread with @JoshLove-msft I considered both the `DelayStrategy` approach during the `WaitUntil.Completed` as well as the custom ResponseClassifier. I agree with Josh and think the custom ResponseClassifier is the better approach to not influence the polling pattern. I have added the custom ResponseClassifier to take into account 404 errors. Based off of the [default parameters of the RetryOptions class](https://github.com/Azure/azure-sdk-for-net/blob/ea7dc0d229a404ba85285ec63fe533a94650f051/sdk/core/Azure.Core/src/RetryOptions.cs#L40), a response resulting in 404 should now be retired 3 times with a delay of 0.8 seconds between each retry.